### PR TITLE
Homework3: Maria Filiptsova

### PR DESCRIPTION
**Pull request by MariaFiliptsova**
### PR checklist
Please make sure you've passed PR checklist
- [ ] my PR passes CI checks (flake8 and pycodestyle)
- [ ] my PR named according to guidelines.
- [ ] my PR contains complete tasks descriptions for all tasks
- [ ] my PR made from the fork feature branch to maxim homework# branch.
- [ ] my PR contains only code that solves appropriate homework tasks. Redundant 
files (configs, etc. ) are put to .gitignore

### CODE REVIEW

File: src/Homework3/ListPractice.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# 1 Используйте генератор списков чтобы получить следующий:
+# ['ab', 'ac', 'ad', 'bb', 'bc', 'bd'].
+spisok = [c + d for c in ['a', 'b'] for d in ['b', 'c', 'd']]
+print(spisok)
+
+# 2 Используйте на предыдущий список slice чтобы получить следующий:
+# ['ab', 'ad', 'bc'].
+sliced_list = spisok[0::2]
```
Comment (belovmd) [inline src/Homework3/ListPractice.py]: redundant 0

**Review by belovmd — COMMENTED** [review]
Thank you for your work. 
Please address next paragraphs from review notes [Review Notes](https://github.com/belovmd/it-academy-summer/blob/master/Homework%20review/Homework3.md):
- 10
- 11
- 16
- 18
- 31
- 33
- 34
- 35

File: src/Homework3/ListPractice.py (hunk 1 above)
Comment (MariaFiliptsova) [reply src/Homework3/ListPractice.py]: very true. It's eliminated

**Comment by MariaFiliptsova** [comment]
All changes according to comments are done.

File: src/Homework3/FizzBuzz.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+но вместо чисел, кратных 3 пишет Fizz,
+вместо чисел кратный 5 пишет Buzz,
+а вместо чисел одновременно кратных и 3 и 5 - FizzBuzz
+'''
+for i in range(1, 101):
+    if not i % 15:
+        print('FizzBuzz')
+    elif i % 3 == 0 and i % 5 != 0:
+        print('Fizz')
+    elif i % 5 == 0 and i % 3 != 0:
```
Comment (belovmd) [inline src/Homework3/FizzBuzz.py:12]: if not i%5 and ... - not addressed from previous review

File: src/Homework3/ListPractice.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+print(spisok)
+
+# 2 Используйте на предыдущий список slice чтобы получить следующий:
+# ['ab', 'ad', 'bc'].
+sliced_list = spisok[::2]
+print(sliced_list)
+
+# 3 Используйте генератор списков чтобы получить следующий
+# ['1a', '2a', '3a', '4a'].
+# spisok2 = [str(c) + d for c in range(1, 5) for d in ['a']]
```
Comment (belovmd) [inline src/Homework3/ListPractice.py:14]: redundant comments

File: src/Homework3/Pairs.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+import math
+
+spisok = '1 2 1 1 1'.split()
+count_appearences = {}
+number_of_pairs = 0
+for i in spisok:
+    count_appearences.update({i: spisok.count(i)})
+for j in count_appearences.values():
+    if j > 1:
+        b_c = math.factorial(j) / (math.factorial(2) * math.factorial(j - 2))
```
Comment (belovmd) [inline src/Homework3/Pairs.py:22]: not O(n)

**Comment by belovmd** [comment]
inaccurate
not all issues addressed
8

**Pull request closed without merging**